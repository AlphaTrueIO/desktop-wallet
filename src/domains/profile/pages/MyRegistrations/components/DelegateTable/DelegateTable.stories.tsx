import React from "react";

import { DelegateTable } from "./DelegateTable";

export default {
	title: "Profile / Pages / My Registrations / Components / Delegate Table",
};

export const Default = () => {
	const registrations = [
		{
			delegate: "OLEBank",
			rank: "#2",
			history: [],
			website: "",
			msq: true,
			repository: [],
		},
		{
			delegate: "OLEBank",
			rank: "#352",
			history: [],
			website: "",
			msq: true,
			repository: [],
		},
	];

	return <DelegateTable data={registrations} />;
};
