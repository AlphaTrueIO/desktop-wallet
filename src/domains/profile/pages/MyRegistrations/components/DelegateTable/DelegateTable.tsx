import { Avatar } from "app/components/Avatar";
import { Button } from "app/components/Button";
import { Circle } from "app/components/Circle";
import { Dropdown } from "app/components/Dropdown";
import { Icon } from "app/components/Icon";
import { Table } from "app/components/Table";
import React from "react";

const columns = [
	{ Header: "Avatar", className: "invisible" },
	{
		Header: "Delegate Name",
		className: "font-semibold text-md",
	},
	{
		Header: "Rank",
		className: "font-semibold text-md -ml-14",
	},
	{
		Header: "MSQ",
		className: "font-semibold text-md",
	},
	{
		Header: "Status",
		className: "font-semibold text-md",
	},
	{
		Header: "Amount",
		className: "invisible w-3 float-right",
	},
	{
		Header: "Votes",
		className: "font-semibold text-md float-right ml-32",
	},
];

type Props = {
	data: any;
};

const options = [
	{ label: "Update", value: "update" },
	{ label: "Transfer", value: "transfer" },
	{ label: "Resign", value: "resign" },
];

const getStatusIcon = (confirmed: boolean) => {
	if (!confirmed) {
		return <Icon name="StatusClock" className="ml-2 text-theme-neutral-500" />;
	}

	return <Icon name="Checkmark" className="ml-1 text-theme-success" width={30} height={30} />;
};

export const DelegateTable = ({ data }: Props) => (
	<div className="flex flex-col bg-theme-background mt-4 px-10">
		<span className="py-10 font-bold text-2xl">Delegate</span>
		<Table columns={columns} data={data}>
			{(rowData: any) => (
				<tr data-testid="delegate-table__row" className="border-b border-dashed border-theme-neutral-light">
					<td className="py-4 w-24">
						<div className="flex items-center">
							<Circle className="border-theme-neutral-800" size="large">
								<Icon name="Business" width={22} height={22} />
							</Circle>
							<Avatar address="APTz" />
						</div>
					</td>
					<td className="py-4 mt-1 font-semibold">
						<span>{rowData.delegate}</span>
					</td>
					<td className="py-1 text-theme-neutral-700">
						<span className="-ml-14">{rowData.rank}</span>
					</td>
					<td className="py-1 text-bold text-theme-primary">
						<Icon name="Msq" width={22} height={22} />
					</td>
					<td className="py-1 text-theme-neutral-400">{getStatusIcon(rowData.confirmed)}</td>
					<td>
						<div className="flex float-right text-semibold text-theme-neutral-800">
							<div className="flex items-center">
								<span className="ml-16">2,450.643</span>
								<Icon name="Ark" className="ml-2" />
							</div>
						</div>
					</td>
					<td className="text-semibold text-theme-neutral-800 w-1/12">
						<div className="flex items-center ml-5 -mr-10">
							<small className="text-theme-neutral-500">2,43%</small>
							<span className="ml-1">2,450.643</span>
							<Icon name="Ark" className="ml-2" />
						</div>
					</td>
					<td className="py-1 float-right">
						<Button variant="plain" className="mt-4" size="small">
							<Dropdown
								toggleIcon="Settings"
								options={options}
								onSelect={(option: any) => console.log(option)}
							/>
						</Button>
					</td>
				</tr>
			)}
		</Table>
	</div>
);
