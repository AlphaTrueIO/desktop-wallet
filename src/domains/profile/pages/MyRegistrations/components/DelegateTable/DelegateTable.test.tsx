import { render } from "@testing-library/react";
import React from "react";

import { DelegateTable } from "./DelegateTable";

describe("Welcome", () => {
	const registrations = [
		{
			agent: "OLEBank",
			delegate: "ARK Ecosystem",
			history: [],
			website: "",
			msq: true,
			repository: [],
		},
		{
			agent: "OLEBank",
			delegate: "ARK Ecosystem",
			history: [],
			website: "",
			msq: true,
			repository: [],
		},
	];

	it("should render empty state", () => {
		const { getAllByTestId, asFragment } = render(<DelegateTable data={registrations} />);

		expect(asFragment()).toMatchSnapshot();
		expect(getAllByTestId("delegate-table__row").length).toEqual(2);
	});
});
