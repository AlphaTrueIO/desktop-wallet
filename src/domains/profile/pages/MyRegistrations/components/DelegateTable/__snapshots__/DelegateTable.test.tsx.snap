// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome should render empty state 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col bg-theme-background mt-4"
  >
    <span
      class="p-10 font-bold text-2xl"
    >
      Delegate
    </span>
    <div
      class="sc-AxhUy fkVaYz"
      role="table"
    >
      <table
        class="table-auto"
      >
        <thead>
          <tr
            role="row"
          >
            <th
              class="text-xs text-left select-none text-theme-neutral-400"
              colspan="1"
              data-testid="table__th--0"
              role="columnheader"
            >
              <div
                class="flex flex-inline align-top  invisible"
              >
                <div
                  class="mr-1"
                >
                  Avatar
                </div>
                <div
                  class="mt-1 text-theme-color-neutral-500"
                  data-testid="table__Sort"
                />
              </div>
            </th>
            <th
              class="text-xs text-left select-none text-theme-neutral-400"
              colspan="1"
              data-testid="table__th--1"
              role="columnheader"
            >
              <div
                class="flex flex-inline align-top  font-semibold text-md ml-5"
              >
                <div
                  class="mr-1"
                >
                  Delegate Name
                </div>
                <div
                  class="mt-1 text-theme-color-neutral-500"
                  data-testid="table__Sort"
                />
              </div>
            </th>
            <th
              class="text-xs text-left select-none text-theme-neutral-400"
              colspan="1"
              data-testid="table__th--2"
              role="columnheader"
            >
              <div
                class="flex flex-inline align-top  font-semibold text-md -ml-20"
              >
                <div
                  class="mr-1"
                >
                  Rank
                </div>
                <div
                  class="mt-1 text-theme-color-neutral-500"
                  data-testid="table__Sort"
                />
              </div>
            </th>
            <th
              class="text-xs text-left select-none text-theme-neutral-400"
              colspan="1"
              data-testid="table__th--3"
              role="columnheader"
            >
              <div
                class="flex flex-inline align-top  font-semibold text-md"
              >
                <div
                  class="mr-1"
                >
                  MSQ
                </div>
                <div
                  class="mt-1 text-theme-color-neutral-500"
                  data-testid="table__Sort"
                />
              </div>
            </th>
            <th
              class="text-xs text-left select-none text-theme-neutral-400"
              colspan="1"
              data-testid="table__th--4"
              role="columnheader"
            >
              <div
                class="flex flex-inline align-top  font-semibold text-md"
              >
                <div
                  class="mr-1"
                >
                  Status
                </div>
                <div
                  class="mt-1 text-theme-color-neutral-500"
                  data-testid="table__Sort"
                />
              </div>
            </th>
            <th
              class="text-xs text-left select-none text-theme-neutral-400"
              colspan="1"
              data-testid="table__th--5"
              role="columnheader"
            >
              <div
                class="flex flex-inline align-top  invisible w-3 float-right"
              >
                <div
                  class="mr-1"
                >
                  Amount
                </div>
                <div
                  class="mt-1 text-theme-color-neutral-500"
                  data-testid="table__Sort"
                />
              </div>
            </th>
            <th
              class="text-xs text-left select-none text-theme-neutral-400"
              colspan="1"
              data-testid="table__th--6"
              role="columnheader"
            >
              <div
                class="flex flex-inline align-top  font-semibold text-md float-right"
              >
                <div
                  class="mr-1"
                >
                  Votes
                </div>
                <div
                  class="mt-1 text-theme-color-neutral-500"
                  data-testid="table__Sort"
                />
              </div>
            </th>
            <th
              class="text-xs text-left select-none text-theme-neutral-400"
              colspan="1"
              data-testid="table__th--7"
              role="columnheader"
            >
              <div
                class="flex flex-inline align-top  invisible float-right"
              >
                <div
                  class="mr-1"
                >
                  Options
                </div>
                <div
                  class="mt-1 text-theme-color-neutral-500"
                  data-testid="table__Sort"
                />
              </div>
            </th>
          </tr>
        </thead>
        <tbody
          role="rowgroup"
        >
          <tr
            class="border-b border-dashed border-theme-neutral-light"
            data-testid="business-table__row"
          >
            <td
              class="py-4"
            >
              <div
                class="ml-10 -mr-16"
              >
                <div
                  class="sc-AxirZ cZokSJ border-theme-neutral-800"
                >
                  <div
                    class="sc-AxiKw fFXzuz"
                    height="22"
                    width="22"
                  >
                    <svg>
                      business.svg
                    </svg>
                  </div>
                </div>
                <div
                  class="sc-AxirZ efFWdY -ml-1"
                />
              </div>
            </td>
            <td
              class="py-4 mt-1 font-semibold pl-5"
            >
              <span />
            </td>
            <td
              class="py-1 text-theme-neutral-700"
            >
              <span
                class="-ml-20"
              />
            </td>
            <td
              class="py-1 text-bold text-theme-primary"
            >
              <div
                class="sc-AxiKw iQZLaj"
                height="25"
                width="25"
              >
                <svg>
                  msq.svg
                </svg>
              </div>
            </td>
            <td
              class="py-1 text-theme-neutral-400"
            >
              <div
                class="sc-AxiKw dzBdqj ml-2 text-theme-neutral-500"
                height="1em"
                width="1em"
              >
                <svg>
                  status-clock.svg
                </svg>
              </div>
            </td>
            <td
              class="px-20 text-bold text-theme-neutral-700"
            >
              <div
                class="flex items-center float-right -mr-20"
              >
                <span>
                  2,450.643
                </span>
                <div
                  class="sc-AxiKw dzBdqj ml-2"
                  height="1em"
                  width="1em"
                >
                  <svg>
                    ark.svg
                  </svg>
                </div>
              </div>
            </td>
            <td
              class="py-1 text-bold text-theme-primary"
            >
              <span
                class="float-right"
              >
                View
              </span>
            </td>
            <td
              class="py-1 float-right"
            >
              <button
                class="sc-AxjAm kTAyCf mt-4"
                color="primary"
                type="button"
              >
                <div
                  class="relative"
                  data-testid="dropdown__toggle"
                >
                  <div
                    class="float-right outline-none focus:outline-none"
                  >
                    <div
                      class="sc-AxiKw dklFyP"
                      height="20"
                      width="20"
                    >
                      <svg>
                        settings.svg
                      </svg>
                    </div>
                  </div>
                </div>
              </button>
            </td>
          </tr>
          <tr
            class="border-b border-dashed border-theme-neutral-light"
            data-testid="business-table__row"
          >
            <td
              class="py-4"
            >
              <div
                class="ml-10 -mr-16"
              >
                <div
                  class="sc-AxirZ cZokSJ border-theme-neutral-800"
                >
                  <div
                    class="sc-AxiKw fFXzuz"
                    height="22"
                    width="22"
                  >
                    <svg>
                      business.svg
                    </svg>
                  </div>
                </div>
                <div
                  class="sc-AxirZ efFWdY -ml-1"
                />
              </div>
            </td>
            <td
              class="py-4 mt-1 font-semibold pl-5"
            >
              <span />
            </td>
            <td
              class="py-1 text-theme-neutral-700"
            >
              <span
                class="-ml-20"
              />
            </td>
            <td
              class="py-1 text-bold text-theme-primary"
            >
              <div
                class="sc-AxiKw iQZLaj"
                height="25"
                width="25"
              >
                <svg>
                  msq.svg
                </svg>
              </div>
            </td>
            <td
              class="py-1 text-theme-neutral-400"
            >
              <div
                class="sc-AxiKw dzBdqj ml-2 text-theme-neutral-500"
                height="1em"
                width="1em"
              >
                <svg>
                  status-clock.svg
                </svg>
              </div>
            </td>
            <td
              class="px-20 text-bold text-theme-neutral-700"
            >
              <div
                class="flex items-center float-right -mr-20"
              >
                <span>
                  2,450.643
                </span>
                <div
                  class="sc-AxiKw dzBdqj ml-2"
                  height="1em"
                  width="1em"
                >
                  <svg>
                    ark.svg
                  </svg>
                </div>
              </div>
            </td>
            <td
              class="py-1 text-bold text-theme-primary"
            >
              <span
                class="float-right"
              >
                View
              </span>
            </td>
            <td
              class="py-1 float-right"
            >
              <button
                class="sc-AxjAm kTAyCf mt-4"
                color="primary"
                type="button"
              >
                <div
                  class="relative"
                  data-testid="dropdown__toggle"
                >
                  <div
                    class="float-right outline-none focus:outline-none"
                  >
                    <div
                      class="sc-AxiKw dklFyP"
                      height="20"
                      width="20"
                    >
                      <svg>
                        settings.svg
                      </svg>
                    </div>
                  </div>
                </div>
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;
